model/DataObject.php

```diff
@@ -423,6 +423,20 @@ private static function cache_composite_fields($class) {
 	}
 
 	/**
+	 * Creates a class instance by the "singleton" design pattern.
+	 * It will always return the same instance for this class,
+	 * which can be used for performance reasons and as a simple
+	 * way to access instance methods which don't rely on instance
+	 * data (e.g. the custom SilverStripe static handling).
+	 *
+	 * @return static The singleton instance
+	 */
+	public static function singleton() {
```

Comment:
I would like this in 3.x, but I'm afraid it wouldn't be semver. XD

The current `singleton()` method on Object is only on the master branch, I'm afraid.